### PR TITLE
MRG, ENH: Publication tweaking example

### LINKED ISSUE #5830: ENH: Example on how to make a figure publication-ready
Every time I go to make a M/EEG figure publication ready, I have to go back to old scripts and dig through StackOverflow to make things presentable. And I've been told by other folks that it's a non-trivial task for them. Basically I want to show people how they can avoid using InkScape / Illustrator.

I propose we add an example that contains a bunch of `matplotlib` tricks to take some MEG data figure and show how you can make it publication-quality. 

## Data content from `sample`

- [x] ERP plot
- [x] Brain activation

## Document structure

- [x]  one block with the standard MNE plots, then a second block with the finalized/tweaked figure ([before/after-style](https://journals.plos.org/ploscompbiol/article/figure?id=10.1371/journal.pcbi.1003833.g004))

## Tricks to include

- Mechanics
   - [x] using `imshow` after a `brain.screenshot()` to show the `mayavi` activation map in a figure
   - [x] adding a `matplotlib` colorbar to match the `mayavi` figure (I have separate code for this that we can integrate)
- Aesthetics
   - [x] removing top and right `spines`
   - [x] adding a legend, tweaking legend properties
   - [x] adding *A* and *B* subplot labels to the upper left of figures
   - [x] changing the number of ticks and their display (e.g., `set(xticks=..., xticklabels=...)`)
   - [x] ticks pointing out
   - [x] setting figure size (`plt.subplots(..., figsize=)`) and finalizing plot properties (`fig.subplots_adjust(...)`)

Others feel free to chime in with little tricks that you have found useful and would like to see added. Eventually I'm hoping what we produce will be pretty enough to be on the front page carousel!

@drammock this might be a good one for when you need a break from digging through tutorials :)

### PR DESCRIPTION
**Pull request by larsoner**
Closes #5830.

@drammock feel free to take over, just wanted to put some of my tweaks down somewhere.

### CODE REVIEW

**Comment by larsoner** [comment]
Not perfect but it's a start

https://15171-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_publication_figure.html#sphx-glr-auto-examples-visualization-plot-publication-figure-py

**Comment by drammock** [comment]
working on some additions; I'll push them this afternoon

**Comment by drammock** [comment]
here's the rendered version: https://15175-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_publication_figure.html#sphx-glr-auto-examples-visualization-plot-publication-figure-py

(latest commit just changes the sphinx thumbnail number)

**Comment by agramfort** [comment]
i let you converge here @larsoner @drammock 

looks very useful !

**Comment by larsoner** [comment]
Okay made a public `mne.viz.plot_brain_colorbar` function (other names welcome, I don't love this one) to simplify the 3D brain plot -> MPL colorbar conversion and tweaked the rendering a bit:

https://15193-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_publication_figure.html#sphx-glr-auto-examples-visualization-plot-publication-figure-py

@drammock feel free to review then maybe @agramfort can merge.

File: mne/utils/docs.py [hunk 1]
```diff
@@ -369,6 +369,49 @@
...
+"""
+docdict["clim_onesided"] = """
+clim : str | dict
+    Colorbar properties specification. If 'auto', set clim automatically
+    based on data percentiles. If dict, should contain:
+
+        ``kind`` : 'value' | 'percent'
+            Flag to specify type of limits.
+        ``lims`` : list | np.ndarray | tuple of float, 3 elements
+            Left, middle, and right bound for colormap.
```
Comment (drammock) [inline mne/utils/docs.py]:
```suggestion
            Lower, middle, and upper bound for colormap.
```

**Comment by drammock** [comment]
rendered: https://15196-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_publication_figure.html#sphx-glr-auto-examples-visualization-plot-publication-figure-py

**Comment by agramfort** [comment]
looks good https://15246-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_publication_figure.html#sphx-glr-auto-examples-visualization-plot-publication-figure-py 

merging. thx @larsoner and @drammock !

**Pull request merged**